### PR TITLE
feat(ComboBox): add selectedItem prop

### LINKED ISSUE #4045: ComboBox `selectedItem` prop like Dropdown
### Summary

`carbon-components-react`'s `ComboBox` component only allows user to set `initialSelectedItem` but will be nice to have `selectedItem` like `Dropdown` component
Please describe your request in one or two sentences.

### Justification

NA

### Desired UX and success metrics

NA

### "Must have" functionality

NA

### Specific timeline issues / requests

NA

### Available extra resources

NA

### PR DESCRIPTION
**Pull request by emyarod**
Closes #4045

This PR adds a `selectedItem` prop to allow for the combobox component to be fully controlled instead of limiting the user to only being able to set the `initialSelectedItem`

#### Changelog

**New**

- `selectedItem` prop (mirrors other listbox component behavior)

#### Testing / Reviewing

Ensure the fully controlled combobox functions as expected ('Add new item' button modifies input and selected item, menu option selection still functions, clear input button still functions)

Ensure the existing uncontrolled combobox contains no regressions

### CODE REVIEW

File: packages/react/src/components/ComboBox/ComboBox-story.js [hunk 1]
```diff
@@ -120,4 +155,9 @@ storiesOf('ComboBox', module)
         text: `Sometimes you want to perform an async action to trigger a backend call on input change.`,
       },
     }
-  );
+  )
+  .add('fully controlled', () => <ControlledComboBoxApp {...props()} />, {
```
Comment (asudoh) [inline packages/react/src/components/ComboBox/ComboBox-story.js]: Probably "application-level control for selection"? Not sure if our combo box allows "fully-controlled" scenario for every key props.

**Review by asudoh — APPROVED** [review]
LGTM basically, just one thing - Thanks @emyarod!

**Review by joshblack — COMMENTED** [review]
Is there any way we could accomplish this instead of adding surface area to ComboBox?

**Comment by emyarod** [comment]
not to my knowledge given this was not included initially unlike the other downshift components, unless you have some ideas?

**Review by vpicone — APPROVED** [review]

**Pull request merged**